### PR TITLE
Add/automated testing

### PR DESCRIPTION
**Pull request by sambua**
## Proposed changes
- Added CI integration with e2e and unit test

## Dependencies
- No

## Checklist
- [x] Tested locally
- [ ] Added new unit, integration or regression tests
- [x] Updated docs

### CODE REVIEW

File: tests/unit/utils/includeSpecs.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+const fs = require('fs');
```
Comment (jimee02) [inline tests/unit/utils/includeSpecs.js:1]: question: Where is this used?

File: .github/actions/install-dependencies/action.yml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        node-version: ${{ inputs.version }}
+        cache: ${{ inputs.pkg-manager }}
+
+    - name: Get npm cache directory
+      id: cache-dir
+      if: ${{ inputs.pkg-manager == 'npm' }}
+      shell: bash
+      run: echo "package-cache-dir=$(npm config get cache)" >> ${GITHUB_OUTPUT}
+
+    - uses: actions/cache@v4
```
Comment (jimee02) [inline .github/actions/install-dependencies/action.yml:48]: comment: Nice! Glad to see support for this in github actions

File: .github/workflows/CI.yml [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+      cache-key: ${{ steps.node.outputs.cache-key }}
+      package-cache-dir: ${{ steps.node.outputs.package-cache-dir }}
+      build-artifact: ${{ steps.build-artifact.outputs.artifact-id }}
+    defaults:
+      run:
+        working-directory: .
+        shell: bash
+    strategy:
+      matrix:
+        node-version: [ '21.x' ]
```
Comment (jimee02) [inline .github/workflows/CI.yml:37]: question: would 22.x make more sense since that'll be long-term support?

File: .github/workflows/CI.yml [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+          pkg-manager: 'npm'
+          use-cache: ${{ inputs.use-cache }}
+          cache-key: ${{ needs.build.outputs.cache-key }}
+
+      - name: Restore build folder
+        uses: ./.github/actions/restore-build-folder
+        with:
+          name: build
+
+      # in the future, we can use different browsers for testing
```
Comment (jimee02) [inline .github/workflows/CI.yml:107]: comment: thanks for preparing for different browsers

File: tests/unit/index.js [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+      value: navOptions[key]
+    });
+  });
+
+  // Below is the polyfill for HTMLMediaElement, as it's not available in JSDOM
+  // this will prevent console the error when running the tests.
+  // But should be careful when using this polyfill, as it's not the real implementation
+  // and may cause the test to pass even if the real implementation is not working.
+  // We have pass these test on real browser via playwright to make sure it's working correctly
+  // if these are really needed we can implement: https://github.com/jsdom/jsdom/issues/2155#issuecomment-581862425
```
Comment (jimee02) [inline tests/unit/index.js:57]: comment: thanks for the backup plan

**Review by jimee02 — APPROVED** [review]
lgtm

File: tests/unit/utils/includeSpecs.js (hunk 1 above)
Comment (sambua) [reply tests/unit/utils/includeSpecs.js:1]: HI @jimee02 , initially I want include specs files manually, after choose other option, and left here as function if could be required in the future, but will remove it for now.

File: .github/workflows/CI.yml (hunk 3 above)
Comment (sambua) [reply .github/workflows/CI.yml:37]: Sure, I can try. On my side it was 21.x which is also not a stable version. Will upgrade it.

**Pull request merged**